tests/baselines/reference/overloadsWithProvisionalErrors.types

```diff
@@ -18,8 +18,8 @@ var func: {
 func(s => ({})); // Error for no applicable overload (object type is missing a and b)
 >func(s => ({})) : never
 >func : { (s: string): number; (lambda: (s: string) => { a: number; b: number; }): string; }
->s => ({}) : (s: string) => {}
->s : string
+>s => ({}) : (s: any) => {}
+>s : any

```

Comment:
These are all places where `string` would get fixed in an earlier overload and bleed into later ones, even though checking for that overload failed.